File: acmp/acmp/1211.py
A, K, B , M, X = map(int, input().split())
L = 1
R = X
while L < R:
    aver = (L + R) // 2
    if (A * (aver - aver // K)) + (B * (aver - aver // M)) < X:
        L = aver + 1
    else:
        R = aver
print(L)

File: acmp/acmp/1211.cpp
#include <iostream>
using namespace std;

int main()
{
    unsigned long long int A, K, B, M, X, R, L, aver;
    cin >> A >> K >> B >> M >> X;
    long double a;
    cin >> a;
    cout << a;
    L = 1;
    R = X;
    while (L < R)
    {
        aver = (L + R) / 2;
        if (aver * (A - A / K) + aver * (B - B / M) < X)
        {
            L = aver + 1;
        }
        else
        {
            R = aver;
        }
    }
    cout << L << endl;
}